Frontend/src/Components/Faults/FaultsView.js

```diff
@@ -1,13 +1,334 @@
-import { Heading, HStack, Text, VStack } from "@chakra-ui/react";
+import { Box, Tooltip, Image, SimpleGrid } from "@chakra-ui/react";
+import MPPTContactorImage from "./Images/MPPT Contactor.png";
+import LowContactorImage from "./Images/Low Contactor.png";
+import MotorControllerContactorImage from "./Images/Motor Controller Contactor.png";
+import DoorImage from "./Images/Door.png";
+import BatteryFailsafeImage from "./Images/Battery Failsafe.png";
+import BPSFaultImage from "./Images/BPS Fault.png";
+import EStopImage from "./Images/E-Stop.png";
+import IMDStatusImage from "./Images/IMD Status.png";
+import CrashImage from "./Images/Crash.png";
+import HighCurrentImage from "./Images/High Current.png";
+import HighVoltageImage from "./Images/High Voltage.png";
+import LowCurrentImage from "./Images/Low Current.png";
+import LowVoltageImage from "./Images/Low Voltage.png";
+import LowBatteryImage from "./Images/Low Battery.png";
+import MCUCheckImage from "./Images/MCU Check.png";
+import MPPTCurrentImage from "./Images/MPPT Current.png";
+import PowerWarningImage from "./Images/Power Warning.png";
+import BMSInputVoltageImage from "./Images/BMS Input Voltage.png";
+import PhysicalConnectionImage from "./Images/Physical Connection Lost.png";
+import WirelessCommsLostImage from "./Images/Wireless Comms Lost.png";
+import HighTemperatureImage from "./Images/High Temperature.png";
 
 export default function Faults(props) {
+    const getEStopString = () => {
+        let eStopStrings = [];
+
+        if(props.data?.battery_eStop[0]) {
+            eStopStrings.push("Battery");
+        }
+        if(props.data?.driver_eStop[0]) {
+            eStopStrings.push("Driver");
+        }
+        if(props.data?.external_eStop[0]) {
+            eStopStrings.push("External");
+        }
+
+        if(eStopStrings.length > 0) {
```

Comment:
The bottom part looks pretty identical to a few other methods, perhaps we can split this off into it's own private function. (iirc private functions in javascript technically don't exist but convention is to preceed them with _ i.e 
```javascript
const myPublicFunction = () => {};
const _myPrivateFunction = () => {};
```